hi-ml-azure/src/health/azure/azure_util.py

```diff
@@ -793,3 +793,64 @@ def is_local_rank_zero() -> bool:
     return global_rank is None and local_rank is None
+
+
+def download_from_datastore(datastore_name: str, prefix: str, output_path: Path,
+                            aml_workspace: Optional[Workspace] = None,
+                            workspace_config_path: Optional[Path] = None,
+                            overwrite: bool = False,
+                            show_progress: bool = False) -> None:
+    """
+    Download a file from an Azure ML Datastore that is registered within a given Workspace.
+
+    If not running inside AML and neither a workspace nor the config file are provided, the code will try to locate a
+    config.json file in any of the parent folders of the current working directory. If that succeeds, that config.json
+    file will be used to create the workspace.
+
+    :param datastore_name: The name of the Datastore containing the blob to be downloaded. This Datastore itself
+        must be an instance of an AzureBlobDatastore.
+    :param prefix: The prefix to the blob to be downloaded
+    :param output_path: The path to which the blob should be downloaded
+    :param aml_workspace: Optional Azure ML Workspace object
+    :param workspace_config_path: Optional path to settings for Azure ML Workspace
+    :param overwrite: If True, will overwrite any existing file at the same remote path
+    :param show_progress: If True, will show the progress of the file download
+    """
+    workspace = get_workspace(aml_workspace=aml_workspace, workspace_config_path=workspace_config_path)
+    datastore = workspace.datastores[datastore_name]
+    assert isinstance(datastore, AzureBlobDatastore), \
+        "Invalid datastore type. Can only download from AzureBlobDatastore"  # for mypy
+    datastore.download(str(output_path), prefix=prefix, overwrite=overwrite, show_progress=show_progress)
+    logging.info(f"Downloaded data to {str(output_path)}")
+
+
+def upload_to_datastore(datastore_name: str, local_data_dir: Path, remote_path: Path,
+                        aml_workspace: Optional[Workspace] = None,
+                        workspace_config_path: Optional[Path] = None,
+                        overwrite: bool = False,
+                        show_progress: bool = False) -> None:
+    """
+    Upload a file to an Azure ML Datastore that is registered within a given Workspace.
+
+    If not running inside AML and neither a workspace nor the config file are provided, the code will try to locate a
+    config.json file in any of the parent folders of the current working directory. If that succeeds, that config.json
+    file will be used to create the workspace.
+
+    :param datastore_name: The name of the Datastore to which the blob should be uploaded. This Datastore itself
+        must be an instance of an AzureBlobDatastore
+    :param local_data_dir: The path to the local directory containing the data to be uploaded
+    :param remote_path: The path to which the blob should be uploaded
```

Comment:
Can you put an example into docu how that works out? What if I have folder "foo" with files "foo/bar/some/file.tsv", and say local_data_dir="foo", "remote_path"="baz/bang", what will my folder structure look like?